scrunch/datasets.py

```diff
@@ -2034,275 +1454,67 @@ def export(self, path, format='csv', filter=None, variables=None,
-                ])
-
-        if not title:
-            title = 'CrunchBox for {}'.format(str(self))
-
-        payload = dict(
-            element='shoji:entity',
-            body=dict(
-                where=variables,
-                filters=filters,
-                force=force,
-                title=title,
-                notes=notes,
-                header=header,
-                footer=footer)
+            expr_obj = expr
+        return self.resource.session.patch(
+            self.resource.fragments.exclusion,
+            data=json.dumps(dict(expression=expr_obj))
         )
 
-        if min_base_size:
-            payload['body'].setdefault('display_settings', {}).update(
-                dict(minBaseSize=dict(value=min_base_size)))
-        if palette:
-            payload['body'].setdefault('display_settings', {}).update(
-                dict(palette=palette))
-
-        # create the boxdata
-        self.resource.boxdata.create(payload)
-
-        # NOTE: the entity from the response is a bit different compared to
-        # others, i.e. no id, no delete method, different entity_url...
-        # For now, return the shoji_tuple from the index
-        for shoji_tuple in self.resource.boxdata.index.values():
-            if shoji_tuple.metadata.title == title:
-                return CrunchBox(shoji_tuple, self)
-
-    def delete_crunchbox(self, **kwargs):
-        """ deletes crunchboxes on matching kwargs """
-        match = False
-        for key in kwargs:
-            if match:
-                break
-            for crunchbox in self.crunchboxes:
-                attr = getattr(crunchbox, key, None)
-                if attr and attr == kwargs[key]:
-                    crunchbox.remove()
-                    match = True
-                    break
-
-
-class DatasetSubvariablesMixin(DatasetVariablesMixin):
-    """
-    Handles a variable subvariables iteration in a dict-like way
-    """
-
-    def __getitem__(self, item):
-        # Check if the attribute corresponds to a variable alias
-        subvariable = self.resource.subvariables.by('alias').get(item)
-        if subvariable is None:
-            # subvariable doesn't exists, must raise a ValueError
-            raise KeyError(
-                'Variable %s has no subvariable with an alias %s' % (self.name,
-                                                                     item))
-        # subvariable exists!, return the subvariable Instance
-        return Variable(subvariable, self)
-
-    def __iter__(self):
-        """
-        Iterable of ordered subvariables
-        """
-        sv_index = self.resource.subvariables.index
-        for sv in self.resource['body']['subvariables']:
-            yield sv_index[sv]
-
-    def __len__(self):
-        return len(self.resource.subvariables.index)
+    def get_exclusion(self):
+        # make sure there is an expression
+        expr = self.resource.exclusion.get('body').get('expression')
```

Comment:
If `body` fails, then this will raise an AttributeError because will return None and None does not have a .get() method. Better to factor this logic to do it by steps.

```
exclusion = self.resource.exclusion
if 'body' not in exclusion:
    return None
expr = exclusion.get('expression')
return prettify(expr, self) if expr else None
```

Also does only 1 request in success case instead of 2